cogs/reward.py

```diff
@@ -11,66 +11,123 @@ def __init__ (self, bot):
+                characterPresent = True
+        
+        if characterPresent == None:
+            tierName = ""
+            if tier.isdigit():
+                tierName = roleArray[int(tier)]
+                tier = tierName
+            else:
+                tierName = tier.capitalize()
+            print(tierName)
+            l = list((re.findall('.*?[hm]', lowerTimeString)))
+            totalTime = 0
+            for timeItem in l:
+                totalTime += convert_to_seconds(timeItem)
 
-        if totalTime == 0:
-            await channel.send(content="You may have formatted the time incorrectly or calculated for 0. Try again with the correct format." + rewardCommand)
-            return
+            if totalTime == 0:
+                await channel.send(content="You may have formatted the time incorrectly or calculated for 0. Try again with the correct format." + rewardCommand)
+                return
 
-        cp = ((totalTime) // 1800) / 2
-        tier = tier.lower()
-        t = 0
-        if tier == 'junior':
-          t = 0
-        elif tier == "journey":
-          t = 1
-        elif tier == "elite":
-          t = 2
-        elif tier == "true":
-          t = 3
-        elif tier == "ascended":
-          t = 4
-        
-        gp = cp* tier_reward_dictionary[t][0]
-        tp = cp* tier_reward_dictionary[t][1]
-        
+            cp = ((totalTime) // 1800) / 2
+            tier = tier.lower()
+            t = 0
+            if tier == 'junior':
+              t = 0
+            elif tier == "journey":
+              t = 1
+            elif tier == "elite":
+              t = 2
+            elif tier == "true":
+              t = 3
+            elif tier == "ascended":
+              t = 4
+            
+            gp = cp* tier_reward_dictionary[t][0]
+            tp = cp* tier_reward_dictionary[t][1]
 
-        treasureArray = [cp, tp, gp]
-        durationString = timeConversion(totalTime)
-        treasureString = f"{treasureArray[0]} CP, {treasureArray[1]} TP, and {treasureArray[2]} GP"
-        await channel.send(content=f"A {durationString} game would give a **{tierName}** Friend\n{treasureString}")
-        return
+            treasureArray = [cp, tp, gp]
+            durationString = timeConversion(totalTime)
+            treasureString = f"{treasureArray[0]} CP, {treasureArray[1]} TP, and {treasureArray[2]} GP"
+            await channel.send(content=f"A {durationString} game would give a **{tierName}** Friend\n{treasureString}")
+            return
+        elif characterPresent == True:
+            l = list((re.findall('.*?[hm]', lowerTimeString)))
+            totalTime = 0
+            for timeItem in l:
+                totalTime += convert_to_seconds(timeItem)
 
+            if totalTime == 0:
+                await channel.send(content="You may have formatted the time incorrectly or calculated for 0. Try again with the correct format." + rewardCommand)
+                return
```

Comment:
This should be extracted, the same check is used for the original case above and I do not see any difference